walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,67 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> busyIntervals = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> freeIntervals = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> importantRanges = new ArrayList<TimeRange>();
+    
+    for(String attendee : request.getAttendees()){
+      for (Event event : events){
+        if(event.getAttendees().contains(attendee)){
+          importantRanges.add(event.getWhen());
+        }
+      }
+    }
+
+    importantRanges.sort(TimeRange.ORDER_BY_START);
+   
+    TimeRange popedRange;
```

Comment:
Rename to `lastBusyRange` or `lastRange` instead of `popedRange`, as there's no stack used here.

Reply:
Eventually I've chosen to use Stack instead of ArrayList (actually ArrayList was used as a Stack so using Stack is more appropriate here). I'm leaving that name for this variable.